code/game/objects/structures/popout_cake.dm

```diff
@@ -159,3 +175,18 @@
 	slices_amount -= Clamp(round(Proj.damage / 3), 1, 4)
 
 	check_slices()
+
+//When spawned, stuffs the corpse underneath it inside
+/obj/structure/popout_cake/corpse_grabber/New()
+	..()
+
+	initialize()
```

Comment:
Yes, but if this cake gets spawned before the object subsystem initializes (if away missions were loaded in the mapping SS), it'll still blow up.